src/interval_sets.jl

```diff
@@ -1,7 +1,22 @@
 
 ###### Set-related Helpers #####
 
-const IntervalSet = AbstractVector{<:AbstractInterval}
+struct IntervalSet{T<:AbstractInterval}
+    items::Vector{<:AbstractInterval}
+end
```

Comment:
```suggestion
struct IntervalSet{T<:AbstractInterval} <: AbstractSet{T}
    items::Vector{T}
end
```

Reply:
I'm not sure what the benefit of subtyping `AbstractSet` is. I feel like these operations are sufficiently different that it might be better to treat it as a set-like object?

Reply:
> I feel like these operations are sufficiently different that it might be better to treat it as a set-like object?

I'd like to dig into this further. In my mind an `IntervalSet` is just a specialize form of a `Set` that understands that an one interval can overlap with another and takes advantage of that property by merging those intervals into one.

Reply:
We can leave making this a subtype of `AbstractSet` for a future PR.

One thought that occurred to me is I'm not sure what your goals for a minimal `IntervalSet` are. My assumption was that `IntervalSet` would gate the new breaking behaviour behind this new type. Possibly you had something different in mind?

Reply:
No, that was my primary goal as well. I'd just like to make as few assumption about it for now. My concern is just that the merging behaviour doesn't really fit any existing set types in the ecosystem is all.

Reply:
I think the problem here is that an `AbstractSet` is a finite set of items and an `Interval` is (theoretically) an infinite set, and an `IntervalSet` represents the union of `Interval`s and is thus also an infinite set. You can iterate an `AbstractSet` but not an `Interval` (side note: I do think we shouldn't implement iteration for `IntervalSet` and should instead pass it off to a function; I would think `IntervalSet` would act more like an `Interval` than it would a collection of `Interval`s.

Reply:
As per @iamed2's suggestion, I've made `IntervalSet`s non-iterable for now. It's a bit uglier in terms of the internals and tests, but it seems more correct for now. We can also make it iterable again later if this is too annoying.